Route feature collections IndexOutOfBoundsException in PrimaryRouteUpdateTask

Robo test reproduced the same 💥 seen in https://github.com/mapbox/mapbox-navigation-android/pull/1735#issue-252800363 after #1735 landed (`master`) 👀 

```
02-13 19:11:26.913: E/AndroidRuntime(17158): Process: com.mapbox.services.android.navigation.app, PID: 17158
02-13 19:11:26.913: E/AndroidRuntime(17158): java.lang.RuntimeException: An error occured while executing doInBackground()
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at android.os.AsyncTask$3.done(AsyncTask.java:304)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at java.lang.Thread.run(Thread.java:818)
02-13 19:11:26.913: E/AndroidRuntime(17158): Caused by: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at java.util.ArrayList.remove(ArrayList.java:403)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at com.mapbox.services.android.navigation.ui.v5.route.PrimaryRouteUpdateTask.doInBackground(PrimaryRouteUpdateTask.java:32)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at com.mapbox.services.android.navigation.ui.v5.route.PrimaryRouteUpdateTask.doInBackground(PrimaryRouteUpdateTask.java:15)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at android.os.AsyncTask$2.call(AsyncTask.java:292)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
02-13 19:11:26.913: E/AndroidRuntime(17158): 	... 4 more
```

--- a/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteLine.java
+++ b/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteLine.java
@@ -304,8 +304,7 @@ class MapRouteLine {
   }
 
   private void updateRoutesFor(int newPrimaryIndex) {
-    boolean isRouteFeatureCollectionsEmpty = routeFeatureCollections.isEmpty();
-    if (newPrimaryIndex < 0 || isRouteFeatureCollectionsEmpty || newPrimaryIndex > routeFeatureCollections.size() - 1) {
+    if (newPrimaryIndex < 0 || newPrimaryIndex > routeFeatureCollections.size() - 1) {
       return;
     }
     new PrimaryRouteUpdateTask(newPrimaryIndex, routeFeatureCollections, primaryRouteUpdatedCallback).execute();

--- a/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/PrimaryRouteUpdateTask.java
+++ b/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/PrimaryRouteUpdateTask.java
@@ -28,6 +28,10 @@ class PrimaryRouteUpdateTask extends AsyncTask<Void, Void, List<FeatureCollectio
   @Override
   protected List<FeatureCollection> doInBackground(Void... voids) {
     List<FeatureCollection> updatedRouteCollections = new ArrayList<>(routeFeatureCollections);
+    if (updatedRouteCollections.isEmpty()) {
+      return routeFeatureCollections;
+    }
+
     // Update the primary new collection
     FeatureCollection primaryCollection = updatedRouteCollections.remove(newPrimaryIndex);
     List<Feature> primaryFeatures = primaryCollection.features();

--- a/libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/route/PrimaryRouteUpdateTaskTest.java
+++ b/libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/route/PrimaryRouteUpdateTaskTest.java
@@ -1,0 +1,27 @@
+package com.mapbox.services.android.navigation.ui.v5.route;
+
+import com.mapbox.geojson.FeatureCollection;
+
+import org.junit.Test;
+
+import java.util.List;
+
+import edu.emory.mathcs.backport.java.util.Collections;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+
+public class PrimaryRouteUpdateTaskTest {
+
+  @Test
+  public void doInBackground_returnsEmptyCollection() {
+    int primaryRouteIndex = 0;
+    List<FeatureCollection> routeFeatureCollections = Collections.emptyList();
+    OnPrimaryRouteUpdatedCallback callback = mock(OnPrimaryRouteUpdatedCallback.class);
+    PrimaryRouteUpdateTask task = new PrimaryRouteUpdateTask(primaryRouteIndex, routeFeatureCollections, callback);
+
+    List<FeatureCollection> updatedCollection = task.doInBackground(mock(Void.class));
+
+    assertEquals(routeFeatureCollections, updatedCollection);
+  }
+}

